htdocs/core/lib/invoice.lib.php

```diff
@@ -456,6 +456,121 @@ function getPurchaseInvoicePieChart($socid = 0)
 	return $result;
 }
 
+/**
+ * Return an HTML table that contains a pie chart of the number of customers or supplier invoices
+ * @param string $mode Can be customer or fourn
+ * @return string A HTML table that contains a pie chart of customers or supplier invoices
+ */
+function getNumberInvoicesPieChart($mode)
+{
+	global $conf, $db, $langs, $user;
+	if (!empty($conf->facture->enabled) && !empty($user->rights->facture->lire)) {
+		include DOL_DOCUMENT_ROOT.'/theme/'.$conf->theme.'/theme_vars.inc.php';
+		$langs->load("boxes");
+		$tmpinvoice = new Facture($db);
+		$sql = "SELECT f.rowid, f.ref, f.fk_statut as status, f.type, f.total_ht, f.total_tva, f.total_ttc, f.paye, f.datef";
```

Comment:
You must use the "due date" of invoice and not the "date" of invoice.

Also imagine there is 1 million of record into database, your alogrithm will return 1 million of record when all what we need is a count. You must something like that: (you must always make the most work as possible by the database engine to reduce network and memory overload).
So use something like that
```
select  
sum( $db->ifsql ( 'date_ech < '.now - 30 days.', 1, 0)) as nbbefore30, 
sum( $db->ifsql ( 'date_ech < '.now - 15 days.', 1, 0)) as nbbefore15, 

etc...
```
You will get immediatly a count for each range in one record answer.